src/main/java/io/codelex/advancedTest/Exercise2/Basket.java

```diff
@@ -0,0 +1,24 @@
+package io.codelex.advancedTest.Exercise2;
+
+public class Basket<T> {
+    private Object[] basket = new Object[10];
+    private int counter = 0;
+
+    public void addToBasket(T item) throws BasketFullException {
+        try {
+            basket[counter] = item;
+            counter++;
+        } catch (RuntimeException e) {
+            throw new BasketFullException("Basket is Full!");
+        }
+    }
+
+    public void removeFromBasket(T item) throws BasketEmptyException {
+        try {
+            basket[counter] = item;
```

Comment:
Man liekas, ka šis īsti nestrādā... Tu vienkārši masīva elementam piešķir jaunu vērtību, bet no masīva jau neko neizņem. Bet šis tiktu atrisināts, ja izmantotu Object[] vietā List<T> vai Stack<T>